netdata/ci: Wrap up coverity script refinement, make it usable, extend CI pipeline compile checks

##### Bug report summary
Currently coverity scans were run exclusively from Travis, resulting on the absence of proper developer toolkit for coverity scanning

As part of this issue we need to:
- [x] wrap up #6747 with a new PR to attend comments and inputs found in #6747

~Update CI pipeline to execute multiple versions of the installer with different configure flags~

- [ ] Make coverity scan run with most (if not all) of the configure flags available for maximum coverage as needed

##### OS / Environment
Any

##### Netdata version (ouput of `netdata -V`)
N/A

##### Component Name
netdata/ci

##### Expected behavior
Coverity should be usable by developers, the CI pipeline that runs coverity should cover all functional cases and once this is resolved C developers should have coverity on their daily agenda

xenstat plugin proved to be more complex that i thought.
We don't seem to have worked with this on distros other than NCP-ng where the initial development took place.

From the tests i did on Ubuntu 18.04 and Cent OS 7 distributions (containerised), i could not get netdata to compile successfully with any of the available packages.

The error i couldn't get through with is:
```
make[2]: Leaving directory `/netdata/web'
make[2]: Entering directory `/netdata'
  CCLD     xenstat.plugin
/tmp/ccm39i63.ltrans0.ltrans.o: In function `vbd_metrics_collect.isra.2':
/netdata/collectors/xenstat.plugin/xenstat_plugin.c:336: undefined reference to `xenstat_vbd_error'
collect2: error: ld returned 1 exit status
make[2]: *** [xenstat.plugin] Error 1
make[2]: Leaving directory `/netdata'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/netdata'
make: *** [all] Error 2
[root@80ea85095f33 netdata]# history
```

After discussing with @vlvkobal, it seems that we need to work with the following repository to get things running for netdata, rather than using an existing binary package. At least i wasn't able to spot which binary package would cover us.

https://github.com/xen-project/xen

--- a/.github/workflows/coverity.yml
+++ b/.github/workflows/coverity.yml
@@ -6,17 +6,16 @@ on:
     - cron: '0 1 * * *'
 jobs:
   coverity:
+    if: github.repository == 'netdata/netdata'
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-        if: github.repository == 'netdata/netdata'
       - name: Prepare environment
-        if: github.repository == 'netdata/netdata'
         run: |
-          bash <(curl -sS https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh) --dont-wait --non-interactive netdata
-          sudo apt-get install -y libjson-c-dev libipmimonitoring-dev libcups2-dev libsnappy-dev libprotobuf-dev libprotoc-dev libssl-dev protobuf-compiler
+          ./packaging/installer/install-required-packages.sh --dont-wait --non-interactive netdata
+          sudo apt-get install -y libjson-c-dev libipmimonitoring-dev libcups2-dev libsnappy-dev \
+                                  libprotobuf-dev libprotoc-dev libssl-dev protobuf-compiler
       - name: Run coverity-scan
-        if: github.repository == 'netdata/netdata'
         run: |
           ./coverity-scan.sh --with-install

--- a/.github/workflows/coverity.yml
+++ b/.github/workflows/coverity.yml
@@ -1,0 +1,22 @@
+---
+# Runs coverity-scan.sh every 24h on `master`
+name: Coverity Scan
+on:
+  schedule:
+    - cron: '0 1 * * *'
+jobs:
+  coverity:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        if: github.repository == 'netdata/netdata'
+      - name: Prepare environment
+        if: github.repository == 'netdata/netdata'
+        run: |
+          bash <(curl -sS https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh) --dont-wait --non-interactive netdata
+          sudo apt-get install -y libjson-c-dev libipmimonitoring-dev libcups2-dev libsnappy-dev libprotobuf-dev libprotoc-dev libssl-dev protobuf-compiler
+      - name: Run coverity-scan
+        if: github.repository == 'netdata/netdata'
+        run: |
+          ./coverity-scan.sh --with-install

--- a/.github/workflows/coverity.yml
+++ b/.github/workflows/coverity.yml
@@ -17,5 +17,8 @@ jobs:
           sudo apt-get install -y libjson-c-dev libipmimonitoring-dev libcups2-dev libsnappy-dev \
                                   libprotobuf-dev libprotoc-dev libssl-dev protobuf-compiler
       - name: Run coverity-scan
+        env:
+          COVERITY_SCAN_TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
+          COVERITY_SCAN_SUBMIT_MAIL: ${{ secrets.COVERITY_SCAN_SUBMIT_MAIL }}
         run: |
           ./coverity-scan.sh --with-install

--- a/.travis.yml
+++ b/.travis.yml
@@ -485,15 +485,6 @@ jobs:
     # This is the nightly pre-execution step (Jobs, preparatory steps for nightly, etc)
     - stage: Nightly operations
 
-      name: Run coverity scan
-      before_script:
-        - bash <(curl -sS https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh) --dont-wait --non-interactive netdata
-        - sudo apt-get install -y libjson-c-dev libipmimonitoring-dev libcups2-dev libsnappy-dev libprotobuf-dev libprotoc-dev libssl-dev protobuf-compiler
-      script: ./coverity-scan.sh --with-install
-      after_failure: post_message "TRAVIS_MESSAGE" "<!here> Coverity nightly run has failed" "${NOTIF_CHANNEL}"
-      env:
-        - ALLOW_SOFT_FAILURE_HERE=true
-
     - name: Kickstart files integrity testing (extended)
       script: ./tests/installer/checksums.sh
 

